on_device_training/mobile/android/java/app/ORTPersonalize/app/build.gradle

```diff
@@ -9,7 +9,7 @@ android {
 
     defaultConfig {
         applicationId "com.example.ortpersonalize"
-        minSdk 21
+        minSdk 23
```

Comment:
https://github.com/microsoft/onnxruntime-training-examples/blob/eb3efa03be87ba3f69b1676de2e47f625ffc5f85/on_device_training/mobile/android/java/app/ORTPersonalize/app/src/main/java/com/example/ortpersonalize/MainActivity.kt#L104-L108

these APIs complained about requiring a minimum of 23